aspnetcore/signalr/java-client.md

```diff
@@ -36,29 +37,35 @@ If using Maven, add the following lines inside the `<dependencies>` element of y
 
 To establish a `HubConnection`, the `HubConnectionBuilder` should be used. The hub URL and log level can be configured while building a connection. Configure any required options by calling any of the `HubConnectionBuilder` methods before `build`. Start the connection with `start`.
 
-[!code-java[Build hub connection](java-client/sample/src/main/java/Chat.java?range=17-20)]
+[!code-java[Build hub connection](java-client/sample/src/main/java/Chat.java?range=16-17)]
 
 ## Call hub methods from client
 
 A call to `send` invokes a hub method. Pass the hub method name and any arguments defined in the hub method to `send`.
 
-[!code-java[send method](java-client/sample/src/main/java/Chat.java?range=31)]
+[!code-java[send method](java-client/sample/src/main/java/Chat.java?range=28)]
 
 ## Call client methods from hub
 
 Use `hubConnection.on` to define methods on the client that the hub can call. Define the methods after building but before starting the connection.
 
-[!code-java[Define client methods](java-client/sample/src/main/java/Chat.java?range=22-24)]
+[!code-java[Define client methods](java-client/sample/src/main/java/Chat.java?range=19-21)]
+
+## Add logging
+
+The SignalR Java client uses the [SLF4J](https://www.slf4j.org/) library for logging. It's a high-level logging API that allows users of the library to chose their own specific logging implementation by bringing in a specific logging dependency. The following code snippet shows how to use `java.util.logging` with the SignalR Java client.
```

Comment:
Put a blurb about the "WARNING" message SLF4J shows when you don't have a binding registered. Make sure to include the warning message verbatim so we get google juice from it.

Reply:
Thanks for that detail. Reminds me - let's presume few of us aside from you and the immediate team have a clue what we're talking about. Helps if we're sharing these that we call out these sorts of details. I'm always afraid to ask, as I'm sure that if I ask a stupid question blowdart will be there with a stamp to put on my forehead [again].